### PR TITLE
client/web: add visual indication for exit node pending approval

### PR DESCRIPTION
**Pull request by mpminardi**
Add visual indication when running as an exit node prior to receiving admin approval.

Updates https://github.com/tailscale/tailscale/issues/10261

https://github.com/tailscale/tailscale/assets/8587567/8b10e1d4-0b43-4aac-bec1-d32d6b392147

### CODE REVIEW

File: client/web/web.go
Comment (soniaappasamy) [inline client/web/web.go:663]: I'd actually send this even if they're not currently advertising as an exit node. That way if already approved, it never shows the error state

**Review by soniaappasamy — COMMENTED** [review]
This is great, very clean! 🚀

File: client/web/web.go
Comment (mpminardi) [reply client/web/web.go:663]: Ah fair, will pull this out and do a `data.AdvertisingExitNodeApproved = routeApproved(v4) || routeApproved(v6)` outside of the loop then 🙏

File: client/web/web.go
Comment (soniaappasamy) [inline client/web/web.go:661]: Actually, this might have to be an `&&` - @willnorris do you know

**Review by soniaappasamy — APPROVED** [review]

File: client/web/web.go
Comment (soniaappasamy) [reply client/web/web.go]: Oh playing with this, I think allowed ips are actually just a subset of the advertised ones 😢

Comment (mpminardi) [reply client/web/web.go]: Ah that is unfortunate 😩

File: client/web/web.go
Comment (willnorris) [reply client/web/web.go:661]: In this case, I honestly don't think it matters.  For the purposes of advertising an exit node, they're always handled together.  I don't think it's possible to have one of them approved without the other.

File: client/web/src/components/exit-node-selector.tsx
Comment (soniaappasamy) [inline client/web/src/components/exit-node-selector.tsx:63]: Nit: add comment

`// refresh data again after short timeout to pick up any pending approval updates`

File: client/web/src/components/exit-node-selector.tsx
Comment (soniaappasamy) [inline client/web/src/components/exit-node-selector.tsx:62]: Could also opt to _just_ do this when `n.ID === runAsExitNode.ID`

**Review by soniaappasamy — COMMENTED** [review]
Looks good, and works consistently for me locally! One nit: it's a little weird to see the pending section pop out after a delay after refreshing the page (we really only care about the delay after togging it). Alessandro had an idea to animate the lower part opening up. That might make it a little more seamless - tailwind has some animation classes that could help us out here.

**Review by soniaappasamy — APPROVED** [review]

**Pull request merged**
